spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mongo/EmbedMongoAutoConfiguration.java

```diff
@@ -0,0 +1,35 @@
+package org.springframework.boot.autoconfigure.mongo;
+
+import com.mongodb.Mongo;
+import de.flapdoodle.embed.mongo.MongodExecutable;
+import de.flapdoodle.embed.mongo.MongodStarter;
+import de.flapdoodle.embed.mongo.config.IMongodConfig;
+import de.flapdoodle.embed.mongo.config.MongodConfigBuilder;
+import de.flapdoodle.embed.mongo.config.Net;
+import de.flapdoodle.embed.mongo.distribution.Version;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.io.IOException;
+
+import static de.flapdoodle.embed.process.runtime.Network.localhostIsIPv6;
+
+@Configuration
+@ConditionalOnClass({ Mongo.class, MongodStarter.class})
+public class EmbedMongoAutoConfiguration {
+
+	@Autowired
+	private MongoProperties properties;
+
+	@Bean(initMethod = "start", destroyMethod = "stop")
+	public MongodExecutable embedMongoServer() throws IOException {
+		IMongodConfig mongodConfig = new MongodConfigBuilder()
+				.version(Version.Main.PRODUCTION)
+				.net(new Net(properties.getPort(), localhostIsIPv6()))
```

Comment:
I think we should think here about something like random port for test environment:
```
@IntegrationTest("spring.data.mongodb.port=0")
```
Something similar to the `server.port=0`:
> Additionally you can set the server.port and management.port properties to 0 in order to run your integration tests using random ports